### PR TITLE
Fix crash with `yield return` as top level statement in script

### PR DESCRIPTION
**Pull request by genlu**
The crash can be reproduced by typing `yield return int.` and try auto-completion (CTRL + space) in a .csx file in VS. This fixes one of the crashes reported in https://github.com/dotnet/roslyn/issues/5390.

@dotnet/interactive

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Binder/InContainerBinder.cs [hunk 1]
```diff
@@ -156,6 +162,19 @@ internal override bool SupportsExtensionMethods
             }
         }
 
+        internal override TypeSymbol GetIteratorElementType(YieldStatementSyntax node, DiagnosticBag diagnostics)
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Binder/InContainerBinder.cs]: Please add a test.

File: src/Compilers/CSharp/Portable/Binder/InContainerBinder.cs [hunk 2]
```diff
@@ -156,6 +162,19 @@ internal override bool SupportsExtensionMethods
...
+        {
+            if (IsScriptClass)
+            {
+                return this.Compilation.GetSpecialType(SpecialType.System_Object);
+            }
+            else
+            {
+                // this path will eventually throw InvalidOperationException
```
Comment (amcasey) [inline src/Compilers/CSharp/Portable/Binder/InContainerBinder.cs]: This path *would* eventually throw, if we didn't have the case above.

File: src/Compilers/CSharp/Portable/Binder/InContainerBinder.cs [hunk 3]
```diff
@@ -156,6 +162,19 @@ internal override bool SupportsExtensionMethods
             }
         }
 
+        internal override TypeSymbol GetIteratorElementType(YieldStatementSyntax node, DiagnosticBag diagnostics)
+        {
+            if (IsScriptClass)
+            {
+                return this.Compilation.GetSpecialType(SpecialType.System_Object);
```
Comment (amcasey) [inline src/Compilers/CSharp/Portable/Binder/InContainerBinder.cs]: Please add a comment explaining why this is here.

**Comment by tannergooding** [comment]
test this please

**Comment by genlu** [comment]
test eta please

**Comment by genlu** [comment]
retest eta please

**Comment by cston** [comment]
LGTM

**Pull request merged**